public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts

```diff
@@ -44,10 +42,13 @@ export default class Datasource extends DataSourceApi<AzureMonitorQuery, AzureDa
     }
 
     if (azureLogAnalyticsOptions.targets.length > 0) {
-      const alaPromise = this.azureLogAnalyticsDatasource.query(azureLogAnalyticsOptions);
-      if (alaPromise) {
-        promises.push(alaPromise);
+      const obs = this.azureLogAnalyticsDatasource.query(azureLogAnalyticsOptions);
+      if (!promises.length) {
+        return obs; // return the observable directly
       }
+      // NOTE: this only includes the data!
+      // When all three query types are ready to be observale, they should all use observable
+      promises.push(obs.toPromise().then(r => r.data));
```

Comment:
This comment is not for this PR but for future work. I think we should consider returning an Observable only and not this mix of both Promise and Observable. The current state makes the code complex.